File: NotificationSystem/NotificationSystem.Web/Controls_Add/ctrAdmin_Add.ascx.cs
using NotificationSystem.NotificationSystem.Data.Classes;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using static NotificationSystem.NotificationSystem.Data.NotificationSystem;

namespace NotificationSystem.NotificationSystem.Web.Controls_Add
{
	public partial class ctrAdmin_Add : System.Web.UI.UserControl
	{

        private string m_UserName = string.Empty;

        public string UserName
        {
            get
            {
                return m_UserName;
            }
            set
            {
                m_UserName = value;
            }
        }

        protected void Page_Load(object sender, EventArgs e)
		{
            clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
            AdminDataTable tblAdmin = new AdminDataTable();

            try
            {
                if ((Page.IsPostBack))
                {
                    if (Request.Form["ctl00$MainContent$ctrAdmin_Add$btnAdd"] == "Add")
                        AddAdmin();
                }
                else
                    PopulateControls();
            }
            catch (Exception)
			{
                throw;
            }
        }

        public void AddAdmin()
        {
            try
            {
                clsAdmin thisAdmin = new clsAdmin();

                {
                    var withBlock = thisAdmin;
                    if (txtusername.Text.Length == 0)
                        return;


                    withBlock.UserName = txtusername.Text; withBlock.Password = txtpassword.Text; withBlock.First = txtfirstname.Text; withBlock.Last = txtlastname.Text; withBlock.Email = txtemailaddress.Text;
                }

                try
                {
                    clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
                    theNotificationSystem.AddAdmin(thisAdmin);
                    lblResult.Text = "Admin data has been added";
                }
                catch (SqlException ex)
                {
                    if (ex.Number == 2627)
                        lblResult.Text = "Admin already exist!";
                    else
                        throw new ApplicationException(ex.Message);
                }
            }


            catch (Exception ex)
            {
                clsNotificationSystem_Web SendError = new clsNotificationSystem_Web();
                string NotificationBody = ex.Message + "  " + ex.StackTrace;
                SendError.SendMailMessage(NotificationBody);
                Response.Redirect("ErrorPage.aspx", false);
            }
        }
        public void PopulateControls()
        {
            clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
        }
        public void ClearControls()
        {
            try
            {

            }


            catch (Exception)
            {
                throw;
            }
        }

        protected void btnCancel_Click(object sender, EventArgs e)
        {
            Response.Redirect("Admin_Find.aspx", false);
        }
    }
}

File: NotificationSystem/NotificationSystem.Web/Controls_Add/ctrAdmin_Add.ascx.vb
Imports System
Imports System.Data.SqlClient
Imports System.Web.UI
Imports NotificationSystem.NotificationSystem.Data.Classes

Namespace NotificationSystem.NotificationSystem.Web.Controls_Add
    Public Partial Class ctrAdmin_Add
        Inherits UserControl

        Private m_UserName As String = String.Empty

        Public Property UserName As String
            Get
                Return m_UserName
            End Get
            Set(value As String)
                m_UserName = value
            End Set
        End Property

        Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

            PopulateControls()
            If Not IsPostBack Then
                txtusername.Text = UserName
                txtusername.Enabled = False
                txtfirstname.Text = ""
                txtlastname.Text = ""
                txtemailaddress.Text = ""
                txtpassword.Text = ""
            End If

        End Sub

        Public Sub AddAdmin()
            Try
                Dim thisAdmin As New clsAdmin

                With thisAdmin

                    .UserName = txtusername.Text
                    .Password = txtpassword.Text
                    .First = txtfirstname.Text
                    .Last = txtlastname.Text
                    .Email = txtemailaddress.Text

                    If txtusername.Text.Length = 0 Then Return

                End With

                Dim theNotificationSystem As New clsNotificationSystem
                    theNotificationSystem.AddAdmin(thisAdmin)
                    lblResult.Text = "Admin data has been added"
                Catch ex As SqlException
                    If ex.Number = 2627 Then
                        lblResult.Text = "Admin already exist!"
                    Else
                        Throw New ApplicationException(ex.Message)
                    End If
                End Try



        End Sub
        Public Sub PopulateControls()
            Dim theNotificationSystem As New clsNotificationSystem
        End Sub
        Public Sub ClearControls()
            Try


            Catch ex As Exception
                Throw
            End Try
        End Sub

        Protected Sub btnCancel_Click(sender As Object, e As EventArgs)
            Response.Redirect("Admin_Find.aspx", False)
        End Sub
    End Class
End Namespace
